app/workers/certificates.php

```diff
@@ -420,25 +422,64 @@ private function notifyError(string $domain, string $errorMessage, int $attempt)
      *
      * @param string $certificateId ID of a new or updated certificate document
      * @param string $domain Domain that is affected by new certificate
+     * @param bool $success Was certificate generation successful?
      *
      * @return void
      */
-    private function updateDomainDocuments(string $certificateId, string $domain): void
+    private function updateDomainDocuments(string $certificateId, string $domain, bool $success): void
     {
-        $domains = $this->dbForConsole->find('domains', [
+        $rule = $this->dbForConsole->findOne('rules', [
             Query::equal('domain', [$domain]),
-            Query::limit(1000),
         ]);
 
-        foreach ($domains as $domainDocument) {
-            $domainDocument->setAttribute('updated', DateTime::now());
-            $domainDocument->setAttribute('certificateId', $certificateId);
-
-            $this->dbForConsole->updateDocument('domains', $domainDocument->getId(), $domainDocument);
-
-            if ($domainDocument->getAttribute('projectId')) {
-                $this->dbForConsole->deleteCachedDocument('projects', $domainDocument->getAttribute('projectId'));
-            }
+        if ($rule !== false && !$rule->isEmpty()) {
+            $rule->setAttribute('certificateId', $certificateId);
+            $rule->setAttribute('status', $success ? 'verified' : 'unverified');
+            $this->dbForConsole->updateDocument('rules', $rule->getId(), $rule);
```

Comment:
doesn't rule have an `updated` attribute anymore?  or are we just using the internal `$updatedAt` attribute ?

Reply:
We use $updatedAt.

Certificate has `updated`. Maybe that's what you are thinking of.